SummaryWriter

Rowan-McKereghan opened the pull request:
- Adding gutenberg-updates summary generation and wikipedia lookup into ebookconverter workflow.
- Needs OpenAI , Anthropic and Serper API keys in .env file to work.
- renovates Autodelete to alllow programmatic invocation
- FileInfo now calls Autodelete.check_book to better handle file deletions and renaming

Rowan-McKereghan committed:
summary writer test

Rowan-McKereghan committed:
commit transaction, pipfile

eshellman commented:
Good start.
1. if you inherit from ebookmaker.writers.TxtWriter, you can pull off the txt directly from the source using the code that makes the txt file. (hint: the text is wrapped in a TxtParser)
2. important to use ORM to load untrusted data. the code I used to load summaries from Johannes is here: https://github.com/JohannesSeikowsky/gutenberg_updates/blob/eric/load_summaries_from_table.py

Rowan-McKereghan committed:
database -> ORM, text from TxtWriter parser

Rowan-McKereghan commented:
replaced database + SQL with ORM, got text from parser (job.url was None running locally, but when I tested with manually set url var, it worked – let me know if I need to debug further)

Rowan-McKereghan committed:
prompts -> txt class py file

eshellman committed:
don't index dot files!

eshellman committed:
0.10.5

eshellman committed:
sychronize with production

eshellman committed:
0.10.6

0.10.5 had packaging issues

eshellman committed:
update requirements (fixes #67)

Rowan-McKereghan committed:
added wikipedia summary scraping before book LLM summary

Rowan-McKereghan committed:
add anthropic to pipfile reqs

Rowan-McKereghan committed:
add wikipedia urls to database, minor prompts change

Rowan-McKereghan committed:
if wiki article already exists, use it

eshellman committed:
Merge pull request #66 from gutenbergtools/v10

0.10.6 has been production since december

eshellman committed:
Merge branch 'master' into pr/65

eshellman committed:
Merge branch 'v11' of https://github.com/Rowan-McKereghan/ebookconverter into pr/65

eshellman committed:
no need to instantiate BeginningBook or FullBook

eshellman committed:
we have a way to test for non-text

eshellman committed:
getting text from parser

eshellman committed:
delint

eshellman committed:
use the book in the job

eshellman committed:
use the summary tags so as not to over-write non-summaries

eshellman committed:
let get_wikipedia_article_summary lnow the book's lang

eshellman committed:
unicode content will still have sentines

eshellman committed:
more docs

eshellman committed:
add AI dependencies, sample env

eshellman committed:
update calls to insert_into_pg_database

eshellman committed:
remove redundant code for already have wiki case.

eshellman committed:
fixed syntax errors, added commits

eshellman committed:
Merge branch 'v11' into v11

eshellman committed:
fix the wikipedia lookup

eshellman committed:
0.11.0

eshellman committed:
Merge branch 'v11' of https://github.com/Rowan-McKereghan/ebookconverter into pr/65

eshellman committed:
remove config, switch to ORM

want to call check_book without the config overhead. GutenbergDatabase.DB has a dependency on config

eshellman committed:
switch to current code style

eshellman committed:
invoke check_book in FileInfo, not in dopush script

fixes #72

eshellman committed:
remove autodelete fro dopush

eshellman committed:
fix duplicate marc bug

There was a constraint violation error when getting wikipedia urls for books with more than one languge wikipedia url.

Needed a fair amount of rearranging code - we pull all the wikipedia urls for a book so we can be sure not to try to save a duplicate.

Also added support for wikipedia urls containing parentheses - actually quite common.

eshellman committed:
0.11.0 final
